.github/workflows/socialsportsprod_cloud_run_deploy_ci.yml

```diff
@@ -0,0 +1,155 @@
+            --entry-point com.functions.events.controllers.CreateEventEndpoint \
+            --runtime java17 \
+            --trigger-http \
+            --allow-unauthenticated \
+            --region australia-southeast1 \
+            --project ${{ matrix.env == 'prod' && 'socialsportsprod' || 'socialsports-44162' }} \
+            --set-env-vars PROJECT_NAME=${{ matrix.env == 'prod' && 'socialsportsprod' || 'socialsports-44162' }}
+
+  deploy-python-firebase-functions:
+    name: Deploy Python Firebase Functions to ${{ matrix.env }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        env: [prod, dev]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Authenticate Google Cloud
+        uses: google-github-actions/auth@v1
+        with:
+          credentials_json: ${{ secrets[matrix.env == 'prod' && 'SOCIALSPORTSPROD_GCLOUD_CREDENTIALS' || 'SOCIALSPORTSDEV_GCLOUD_CREDENTIALS'] }}
+
+      - name: Install Firebase CLI
+        run: |
+          npm install -g firebase-tools
+
+      - name: Setup Python 3.12
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.12
+
+      - name: Set up Python environment
+        working-directory: functions
+        run: |
+          python3.12 -m venv venv
+          source venv/bin/activate
+          pip install -r requirements.txt
+
+      - name: Create functions_key.json from base64-encoded secret
+        working-directory: functions
+        run: |
+          echo '${{ secrets[matrix.env == 'prod' && 'SOCIALSPORTSPROD_FUNCTIONS_KEY_JSON_BASE64_ENCODED' || 'SOCIALSPORTSDEV_FUNCTIONS_KEY_JSON_BASE64_ENCODED'] }}' | base64 -d > functions_key.json
+
+      - name: Set .env variables
+        working-directory: functions
+        run: |
+          echo "STRIPE_API_KEY=\"${{ secrets[matrix.env == 'prod' && 'SOCIALSPORTSPROD_STRIPE_API_KEY' || 'SOCIALSPORTSDEV_STRIPE_API_KEY'] }}\"" >> .env
+          echo "STRIPE_WEBHOOK_ENDPOINT_SECRET=\"${{ secrets[matrix.env == 'prod' && 'SOCIALSPORTSPROD_STRIPE_WEBHOOK_ENDPOINT_SECRET' || 'SOCIALSPORTSDEV_STRIPE_WEBHOOK_ENDPOINT_SECRET'] }}\"" >> .env
+          echo "POSTHOG_API_KEY=\"${{ secrets[matrix.env == 'prod' && 'SOCIALSPORTSPROD_POSTHOG_API_KEY' || 'SOCIALSPORTSDEV_POSTHOG_API_KEY'] }}\"" >> .env
+          echo "BEARER_TOKEN=\"${{ secrets[matrix.env == 'prod' && 'SOCIALSPORTSPROD_BEARER_TOKEN' || 'SOCIALSPORTSDEV_BEARER_TOKEN'] }}\"" >> .env
+          echo "SENDGRID_API_KEY=\"${{ secrets[matrix.env == 'prod' && 'SOCIALSPORTSPROD_SENDGRID_API_KEY' || 'SOCIALSPORTSDEV_SENDGRID_API_KEY'] }}\"" >> .env
+
+      - name: Set firebase project
+        working-directory: functions
+        run: |
+          firebase use --add ${{ matrix.env == 'prod' && 'socialsportsprod' || 'socialsports-44162' }}
+
+      - name: Deploy all python functions
+        working-directory: functions
+        run: |
+          firebase deploy --only functions:move_inactive_events
```

Comment:
Currently, I am only deploying move_inactive_events, so critical pathways and functions (e.g., stripe payment logic) is not affected in case of faulty deployments. Once the logic has been reviewed by others and approved, I will add the rest of the firebase functions to be auto deployed.